src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1019,6 +1038,23 @@ internal bool CanFocusSearch(object parameter)
             return true;
         }
 
+        internal void ToggleLayout(object parameter)
```

Comment:
You can use a converter in this case. Attach a converter in XAML, which binds to IsDetailedMode. Any time, only one of them can be true. 
OR,  you can pass true or false in command parameter. and say IsDetailedMode = (bool) parameter. Just one line.